aws/templates/application/application_apigateway.ftl

```diff
@@ -424,6 +424,125 @@
+                                            "docs")]
+
+                [#assign docsS3WebsiteConfiguration = getS3WebsiteConfiguration("apidoc.html", "")]
+
+                [#assign docsS3BucketName = (occurrence.DNSIsConfigured && occurrence.DNS.Enabled)?then(
+                                                formatDomainName(
+                                                    occurrence.Publish.DnsNamePrefix,
+                                                    dns
+                                                ),
+                                                formatName(
+                                                    occurrence.Publish.DnsNamePrefix,
+                                                    tier,
+                                                    component,
+                                                    occurrence
+                                                )
+                                                )]    
+                
+                [#assign docsWAFCIDRList = [] ]
+
+                [#if occurrence.WAFIsConfigured &&
+                        occurrence.WAF.Enabled &&
+                        ipAddressGroupsUsage["waf"]?has_content ]
+                    
+                    [#list occurrence.WAF.IPAddressGroups as group]
+                            
+                            [#assign groupId = group?is_hash?then(
+                                            group.Id,
+                                            group)]
+                            
+                            [#if (ipAddressGroupsUsage["waf"][groupId])?has_content]
+                                
+                                [#assign docsWAFCIDRList +=  (ipAddressGroupsUsage["waf"][groupId]).CIDR  ]
+                                
+                            [/#if]
+                            
+                    [/#list]
+                    
+                [/#if]
+                
+                [#assign docsS3IPWhitelist = (docsWAFCIDRList?has_content)?then(
+                                                s3IPAccessCondition(docsWAFCIDRList),
+                                                "*")]                  
+
+                [@createBucketPolicy
+                    mode=listMode
+                    id=docsS3BucketPolicyId
+                    bucket=docsS3BucketId
+                    statements=
+                        s3ReadPermission(
+                            docsS3BucketName,
+                            "",
+                            "*",
+                            "*",
+                            docsS3IPWhitelist
+                        )
+                    dependencies=docsS3BucketId
+                /]
+
+                [@createS3Bucket
+                    mode=listMode
+                    id=docsS3BucketId
+                    name=docsS3BucketName
+                    websiteConfiguration=docsS3WebsiteConfiguration
+                    outputId=docsS3BucketId
```

Comment:
outputId is really only there for legacy support of some situations where the id used for outputs was different to that used to define the resource. Should always have been the same. In this case they are so the outputId isn't needed